$NetBSD: patch-configure.ac,v 1.10 2016/02/13 08:46:03 wiz Exp $

* Use gstreamer under OpenBSD
* Add FreeBSD support (GNU patch is not needed)
* Detect pkgsrc apache-ant library
* Fix installation directory
* Adapt for mdds-1.0.

--- configure.ac.orig	2015-11-25 12:55:38.000000000 +0000
+++ configure.ac
@@ -713,6 +713,8 @@ aix*)
 
 openbsd*)
     test_gtk=yes
+    build_gstreamer=yes
+    build_gstreamer_0_10=yes
     test_tde=yes
     test_kde=yes
     test_freetype=yes
@@ -6212,7 +6214,7 @@ if test \( "$_os" != "WINNT" -o "$WITH_M
 
     # Assume visibility is not broken with libc++. The below test is very much designed for libstdc++
     # only.
-    if test "$HAVE_GCC_VISIBILITY_FEATURE" = "TRUE" -a "$CPP_LIBRARY" = GLIBCXX; then
+    if test "$HAVE_GCC_VISIBILITY_FEATURE" = "TRUE" -a "$CPP_LIBRARY" = GLIBCXX -a "$COM_GCC_IS_CLANG" != "TRUE"; then
         dnl gcc#19664, gcc#22482, rhbz#162935
         AC_MSG_CHECKING([if STL headers are visibility safe (GCC bug 22482)])
         AC_EGREP_HEADER(visibility push, string, stlvisok=yes, stlvisok=no)
@@ -6704,7 +6706,7 @@ dnl ====================================
 dnl Set target Java bytecode version
 dnl ===================================================================
 if test "$ENABLE_JAVA" != ""; then
-    if test "$HAVE_JAVA9" == "TRUE"; then
+    if test "$HAVE_JAVA9" = "TRUE"; then
         _java_target_ver="1.6"
     else
         _java_target_ver="1.5"
@@ -6764,30 +6766,7 @@ dnl ====================================
 dnl Checks for javadoc
 dnl ===================================================================
 if test "$ENABLE_JAVA" != ""; then
-    if test -z "$with_jdk_home"; then
-        AC_PATH_PROG(JAVADOC, javadoc)
-    else
-        _javadoc_path="$with_jdk_home/bin/javadoc"
-        dnl Check if there is a javadoc at all.
-        if test -x "$_javadoc_path"; then
-            JAVADOC=$_javadoc_path
-        else
-            AC_PATH_PROG(JAVADOC, javadoc)
-        fi
-    fi
-    if test -z "$JAVADOC"; then
-        AC_MSG_ERROR([$_javadoc_path not found set with_jdk_home])
-    fi
-    if test "$build_os" = "cygwin"; then
-        if test x`echo "$JAVADOC" | $GREP -i '\.exe$'` = x; then
-            JAVADOC="${JAVADOC}.exe"
-        fi
-        JAVADOC=`win_short_path_for_make "$JAVADOC"`
-    fi
-
-    if test `$JAVADOC --version 2>&1 | $GREP -c "gjdoc"` -gt 0; then
-    JAVADOCISGJDOC="yes"
-    fi
+    AC_PATH_PROG(JAVADOC, javadoc)
 fi
 AC_SUBST(JAVADOCISGJDOC)
 
@@ -7789,7 +7768,7 @@ no|disable)
     else
         # Unset variables set by the above AM_PATH_PYTHON so that
         # we actually do check anew.
-        unset PYTHON am_cv_pathless_PYTHON ac_cv_path_PYTHON am_cv_python_version am_cv_python_platform am_cv_python_pythondir am_cv_python_pyexecdir
+        #unset PYTHON am_cv_pathless_PYTHON ac_cv_path_PYTHON am_cv_python_version am_cv_python_platform am_cv_python_pythondir am_cv_python_pyexecdir
         AM_PATH_PYTHON([3.3],, [:])
         if test "$PYTHON" = ":"; then
             if test -z "$PYTHON_FOR_BUILD"; then
@@ -7842,9 +7821,9 @@ if test $enable_python = system; then
     elif test "$cross_compiling" != yes; then
         # Unset variables set by the above AM_PATH_PYTHON so that
         # we actually do check anew.
-        unset PYTHON am_cv_pathless_PYTHON ac_cv_path_PYTHON am_cv_python_version am_cv_python_platform am_cv_python_pythondir am_cv_python_pyexecdir
+        #unset PYTHON am_cv_pathless_PYTHON ac_cv_path_PYTHON am_cv_python_version am_cv_python_platform am_cv_python_pythondir am_cv_python_pyexecdir
         # This causes an error if no python command is found
-        AM_PATH_PYTHON([3.3])
+        AM_PATH_PYTHON([2.6])
         python_include=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('INCLUDEPY'));"`
         python_version=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('VERSION'));"`
         python_libs=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('LIBS'));"`
@@ -8528,7 +8507,7 @@ AC_SUBST(SYSTEM_BOOST)
 dnl ===================================================================
 dnl Check for system mdds
 dnl ===================================================================
-libo_CHECK_SYSTEM_MODULE([mdds], [MDDS], [mdds >= 0.11.2], ["-I${WORKDIR}/UnpackedTarball/mdds/include"])
+libo_CHECK_SYSTEM_MODULE([mdds], [MDDS], [mdds-1.0 >= 0.11.2], ["-I${WORKDIR}/UnpackedTarball/mdds/include"])
 
 dnl ===================================================================
 dnl Check for system glm
@@ -9688,8 +9667,8 @@ if test -z "$PATCH"; then
     AC_MSG_ERROR(["patch" not found in \$PATH, install it])
 fi
 
-dnl On Solaris, FreeBSD or MacOS X, check if --with-gnu-patch was used
-if test "$_os" = "SunOS" -o "$_os" = "FreeBSD" -o "$_os" = "Darwin"; then
+dnl On Solaris or MacOS X, check if --with-gnu-patch was used
+if test "$_os" = "SunOS" -o "$_os" = "Darwin"; then
     if test -z "$with_gnu_patch"; then
         GNUPATCH=$PATCH
     else
@@ -11968,7 +11947,7 @@ EOF
     dnl Checking for ant.jar
     if test "$ANT_HOME" != "NO_ANT_HOME"; then
         AC_MSG_CHECKING([Ant lib directory])
-        if test -f $ANT_HOME/lib/ant.jar; then
+        if test -f $ANT_HOME/lib/java/ant/ant.jar; then
             ANT_LIB="$ANT_HOME/lib"
         else
             if test -f $ANT_HOME/ant.jar; then
@@ -12389,7 +12368,7 @@ AC_SUBST(MANDIR)
 DOCDIR=[$(eval echo $(eval echo $docdir))]
 AC_SUBST(DOCDIR)
 
-INSTALLDIR="$LIBDIR/$INSTALLDIRNAME"
+INSTALLDIR="$PREFIXDIR"
 AC_SUBST(INSTALLDIR)
 
 TESTINSTALLDIR="${BUILDDIR}/test-install"
